lib/BackbeatConsumer.js

```diff
@@ -783,9 +784,31 @@ class BackbeatConsumer extends EventEmitter {
         }
         this._circuitBreaker.stop();
         return async.series([
+            next => {
+                // stopping consumption of new messages while
+                // waiting for current ones to be processed
+                this._isShuttingDown = true;
+                // waiting for all current jobs to end before exiting
+                if (this._processingQueue && !this._processingQueue.idle()) {
+                    this._processingQueue.drain = () => {
+                        this._log.info('drained processing queue', {
+                            method: 'BackbeatConsumer.close',
+                        });
+                        next();
+                    };
+                    return;
```

Comment:
not needed (there is no other return here)

Reply:
if i don't return here `process.nextTick(next);` will be called and we won't wait for the drain